pkg/controller/account/account_controller.go

```diff
@@ -605,6 +604,27 @@ func CreateAccount(reqLogger logr.Logger, client awsclient.Client, accountName,
 	return accountStatus, nil
 }
 
+func checkAWSAccountsLimitReached(r *ReconcileAccount, reqLogger logr.Logger, currentAccounts int) (bool, error) {
+	instance := &corev1.ConfigMap{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Namespace: awsv1alpha1.AccountCrNamespace, Name: awsv1alpha1.DefaultConfigMap}, instance)
+	if err != nil {
+		unexpectedErrorMsg := fmt.Sprintf("%s: Failed to retrieve default ConfigMap, account limit defaulting to 100", awsv1alpha1.ErrMissingDefaultConfigMap)
+		reqLogger.Info(unexpectedErrorMsg)
+	} else {
```

Comment:
I'm not a huge fan of this giant else block (or else statements 98% of the time, really).  This is pretty confusing to read and understand, at least at first glance.  Is there any way we could just return more often rather than rely on else statements through here?

Reply:
@iamkirkbater If you think it would be appropriate to add comments I'm happy to do that but as far as the control flow goes I don't know how it can be any more efficient or clean. If you have any suggestions I'm all ears.

Reply:
You know, maybe it was the fact that I was looking at this in the diff page and it had some of the comments in it so it looked a lot longer than I originally thought, when I went to the actual file to look at this it doesn't look as bad.  I can live with this.  If we need to add another else in here, though, I'm going to suggest refactoring again, but we shouldn't do that until there's a few unit tests in place anyway.